lib/at_shell/at_shell.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+
+#include <shell/shell.h>
+#include <nrf_modem_at.h>
+#include <modem/at_monitor.h>
+
+static struct shell *global_shell;
+static const char at_usage_str[] =
+	"Usage: at <subcommand>\n"
+	"\n"
+	"Subcommands:\n"
+	"  events_enable     Enable AT event handler which prints AT notifications\n"
+	"  events_disable    Disable AT event handler\n"
+	"\n"
+	"Any other subcommand is interpreted as AT command and sent to the modem.\n";
+
+AT_MONITOR(at_shell_monitor, ANY, at_notif_handler, PAUSED);
+
+static void at_notif_handler(const char *response)
+{
+	shell_print(global_shell, "AT event handler:\n%s", response);
```

Comment:
As this is the only place where `global_shell` is used, maybe we could hide it's use into this function by using `shell_backend_uart_get_ptr()`.
Or, can we have some other than UART backed?

Reply:
I think RTT backend would also be used by some